sensor/common/networkflow/manager/manager_impl.go

```diff
@@ -410,6 +524,26 @@ func (m *networkFlowManager) enrichHostContainerEndpoints(hostConns *hostConnect
 	flowMetrics.HostEndpointsRemoved.Add(float64(prevSize - len(hostConns.endpoints)))
 }
 
+func (m *networkFlowManager) enrichProcessesListening(hostConns *hostConnections, processesListening map[processListeningIndicator]timestamp.MicroTS) {
+	hostConns.mutex.Lock()
```

Comment:
I wonder if this could be done with a `RWMutex` to avoid contention since you're not changing any values in the `hostConnections` object.

Reply:
I honestly just copy pasted it from the existing `enrichContainerEndpoints`, my understanding is the only 2 threads working on this are the one receiving messages from collector and this one processing them and pushing them up the chain. If that's the case, I don't think we would benefit much from using a `RWMutex` since whichever one grabs it will lock out the other one anyways.

Reply:
Yes. But from what I've seen, neither of them change the object under lock. Which means that if you used a `RWMutex` they won't lock each other for read-only operations. Maybe I'm missing something obvious?

You don't need to do anything here. I can investigate this further. This was more of a highlighting a point of improvement for us.

Reply:
It's not obvious but AFAICT, the three enrich functions here will run sequentially and the only place this mutex is used seems to be in `hostConnections.Process` which gets called from a separate routine here: https://github.com/stackrox/stackrox/blob/dafad863cffd05d21643e8ef22a498835be944e6/sensor/common/networkflow/service/service_impl.go#L137

So we should be back to no real gain using a `RWLock` until we decide to move one of the enrich functions to a separate routine.